pnut.c

```
@@ -2361,6 +2389,11 @@ ast parse_definition(int local) {
   ast tail = 0;
   ast current_declaration;
 
+  //Currently, static is just skipped
+  if(tok == STATIC_KW) {
+    get_tok();
+  }
```

I don't think we want the parser to silently drop the `static` specifier since it changes the semantic of the program (as opposed to `const`, where dropping the `const` from a valid C program doesn't change its semantic).

If you want pnut to ignore the `static` keywords, I suggest you add `define static` at the top of your project so it expands to empty.

---

Ah, I meant to only skip the static keyword when it appeared in front of functions. I didn't realize `parse_definition` wasn't only for functions. 

I think(?) skipping static for functions should be fine since static doesn't change semantics for functions, it just prevents them from being available in other compilation units (and as far as I can tell, pnut only supports one compilation unit). 

Actually for global definitions, static shouldn't change semantics because the definitions are already file scoped. static just prevents them from being exported when linking. 

If I am correct about pnut only supporting a single compilation unit, skipping `static` in global definitions shouldn't affect semantics

---

I agree! The `local` argument of `parse_definition` indicates if the definition is local or global so it should a pretty easy to implement.